lib/ansible/modules/database/postgresql/postgresql_privs.py

```diff
@@ -679,9 +727,15 @@ def build(self):
 
     def add_default_revoke(self):
         for obj in self._objs:
-            self.query.append(
-                'ALTER DEFAULT PRIVILEGES IN SCHEMA {0} REVOKE ALL ON {1} FROM {2};'.format(self._schema, obj,
-                                                                                            self._for_whom))
+            if not self._as_who:
```

Comment:
I would swap the conditionals: `if self._as_who: ... else: `.

Thinking in the current `else` branch is kind of weird: `not not self._as_who`